### PR TITLE
Post-release 0.1.0

### LINKED ISSUE #4: Add identifiers field to CITATION.cff after first release
We need the first release to generate a DOI on [Zenodo](https://zenodo.org/).

```yaml
identifiers:
  - description:
    type: doi
    value:
```

### PR DESCRIPTION
**Pull request by RMeli**
* Close #4
* Support HTTPS as `git` operations protocol
* Add check on `git status` (no un-commited changes allowed when running `roll_release.sh`)
* Add version `0.1.0` to (local) Spack package

### CODE REVIEW

**Comment by rasolca** [comment]
I would add to README [zenodo](https://doi.org/10.5281/zenodo.11241331)
`[zenodo](https://doi.org/10.5281/zenodo.11241331)`

and possibly even the status of the pipeline [pipeline status](https://gitlab.com/cscs-ci/ci-testing/webhook-ci/mirrors/657496524998283/7598378243915359/-/commits/main)

`[pipeline status](https://gitlab.com/cscs-ci/ci-testing/webhook-ci/mirrors/657496524998283/7598378243915359/-/commits/main)`

**Comment by RMeli** [comment]
Excellent suggestion! Should I also add the CSCS logo or other acknowledgements?

**Comment by rasolca** [comment]
> Excellent suggestion! Should I also add the CSCS logo or other acknowledgements?

Right, I would add an acknowledgment section as in DLA-Future with CSCS, ETHZ and PASC.

File: spack/packages/dla-future-fortran/package.py [hunk 1]
```diff
@@ -16,11 +16,12 @@ class DlaFutureFortran(CMakePackage):
     url = "https://github.com/eth-cscs/DLA-Future-Fortran/archive/v0.0.0.tar.gz"
     git = "https://github.com/eth-cscs/DLA-Future-Fortran.git"
 
-    maintainers("RMeli", "rasolca")
+    maintainers("RMeli", "rasolca", "aurianer")
 
     license("BSD-3-Clause")
 
     version("main")
```
Comment (RMeli) [inline spack/packages/dla-future-fortran/package.py:23]:
```suggestion
    version("main", branch="main")
```

**Review by msimberg — APPROVED** [review]

**Review by rasolca — APPROVED** [review]

**Comment by RMeli** [comment]
Spack is acting up... I'll need to look into it more carefully.

**Review by msimberg — APPROVED** [review]

**Comment by RMeli** [comment]
> Spack is acting up... I'll need to look into it more carefully.

With the build cache enabled, concretisation ends up having the same packages multiple times with different `arch` (`ubuntu20.04`/`ubuntu22.04`).

**Comment by RMeli** [comment]
Blocked by https://github.com/eth-cscs/DLA-Future-Fortran/pull/9.

**Pull request merged**